pkg/snapstore/snapstore_test.go

```diff
@@ -278,10 +280,271 @@ var _ = Describe("Save, List, Fetch, Delete from mock snapstore", func() {
 	})
 })
 
-func resetObjectMap() {
-	for k := range objectMap {
-		delete(objectMap, k)
+// directory where the credential files created are stored
+const (
+	credentialDirectory = "../../test/credential"
+)
+
+var (
+	// environment variables for each provider
+	providers = []string{
+		"AWS_APPLICATION_CREDENTIALS",
+		"AZURE_APPLICATION_CREDENTIALS",
+		"OPENSTACK_APPLICATION_CREDENTIALS", // V3ApplicationCredentials
+		"OPENSTACK_APPLICATION_CREDENTIALS", // Password
+		"ALICLOUD_APPLICATION_CREDENTIALS",
+		"OPENSHIFT_APPLICATION_CREDENTIALS",
+	}
+	credentialFilesForProviders = [][]string{
+		{"accessKeyID", "region", "secretAccessKey"},
+		{"storageAccount", "storageKey"},
+		{"authURL", "tenantName", "domainName", "applicationCredentialID", "applicationCredentialName", "applicationCredentialSecret"},
+		{"authURL", "tenantName", "domainName", "username", "password"},
+		{"accessKeyID", "accessKeySecret", "storageEndpoint"},
+		{"accessKeyID", "region", "endpoint", "secretAccessKey"},
+	}
+	providerToSnapstoreProviderMap = map[string]string{
+		"AWS_APPLICATION_CREDENTIALS":       brtypes.SnapstoreProviderS3,
+		"AZURE_APPLICATION_CREDENTIALS":     brtypes.SnapstoreProviderABS,
+		"OPENSTACK_APPLICATION_CREDENTIALS": brtypes.SnapstoreProviderSwift,
+		"ALICLOUD_APPLICATION_CREDENTIALS":  brtypes.SnapstoreProviderOSS,
+		"OPENSHIFT_APPLICATION_CREDENTIALS": brtypes.SnapstoreProviderOCS,
+	}
+	providerToSnapstoreProviderJSONMap = map[string]string{
+		"AWS_APPLICATION_CREDENTIALS_JSON":       brtypes.SnapstoreProviderS3,
+		"GOOGLE_APPLICATION_CREDENTIALS":         brtypes.SnapstoreProviderGCS,
+		"AZURE_APPLICATION_CREDENTIALS_JSON":     brtypes.SnapstoreProviderABS,
+		"OPENSTACK_APPLICATION_CREDENTIALS_JSON": brtypes.SnapstoreProviderSwift,
+		"ALICLOUD_APPLICATION_CREDENTIALS_JSON":  brtypes.SnapstoreProviderOSS,
+		"OPENSHIFT_APPLICATION_CREDENTIALS_JSON": brtypes.SnapstoreProviderOCS,
+	}
+)
+
+var _ = Describe("Dynamic access credential rotation for each provider", func() {
+	// Credentials in a directory
+	for providerIndex, provider := range providers {
+		// value is used in the closure, thus is to be redefined
+		providerIndex := providerIndex
+		provider := provider
```

Comment:
The reason the loop variables, and many variables in the tests had to be redefined as so was that the closure functions which are used to define the test cases using the Ginkgo framework capture the address of the loop variables when `Describe`, `Context`, and `It` functions run. This is essentially the infamous `for` loop "bug" when using the address of the loop variable inside the `for` loop in Go. Redefining ensures the right values are passed to the closures in new variables instead of the actual loop variable which is rewritten every time during the spec tree construction, before the tests are actually run.